src/main/java/pl/debuguj/parkingspacessystem/controllers/ParkingSpaceController.java

```diff
@@ -0,0 +1,142 @@
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectDateException;
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectEndDateException;
+import pl.debuguj.parkingspacessystem.exceptions.ParkingSpaceNotFoundException;
+import pl.debuguj.parkingspacessystem.services.ParkingSpaceManagementService;
+
+import java.math.BigDecimal;
+import java.util.Date;
+
+/**
+ * Created by grzesiek on 07.10.17.
+ */
+@RestController
+public class ParkingSpaceController {
+
+    private static final Logger logger = LoggerFactory.getLogger(ParkingSpaceController.class);
+
+    public static final String URI_START_METER = "/startMeter";
+    public static final String URI_CHECK_VEHICLE = "/checkVehicle";
+    public static final String URI_STOP_METER = "/stopMeter";
+    public static final String URI_CHECK_INCOME_PER_DAY = "/checkIncomePerDay";
+
+    private final ParkingSpaceManagementService parkingSpaceManagement;
+    private final Constants constants;
+
+    public ParkingSpaceController(ParkingSpaceManagementService parkingSpaceManagement, Constants constants) {
+        this.parkingSpaceManagement = parkingSpaceManagement;
+        this.constants = constants;
+    }
+
+    @PostMapping( value = URI_START_METER + "/{registrationNumber:[0-9]{5}}" )
+    public HttpEntity<?> startParkingMeter(
+            @PathVariable() final String registrationNumber,
+            @RequestParam() final DriverType driverType,
+            @RequestParam() final String startTime,
+            @RequestParam() final String stopTime)
+                throws IncorrectEndDateException, CarRegisteredInSystemException, IncorrectDateException
+    {
+        Date begin = validateDate(startTime, constants.getTimeFormat());
+        Date end = validateDate(stopTime, constants.getTimeFormat());
+
+        if(begin != null && end != null)
+        {
+            ParkingSpace ps = new ParkingSpace(registrationNumber, begin, end);
+            ps.setDriverType(driverType);
+
+            return new ResponseEntity<>( parkingSpaceManagement.reserveParkingSpace(ps), HttpStatus.OK);
+        }
+        else {
+            throw new IncorrectDateException("Incorrect date format");
+        }
+    }
+
+    @GetMapping(value = URI_CHECK_VEHICLE + "/{registrationNumber:[0-9]{5}}")
+    public HttpEntity<?> checkVehicle(
+            @PathVariable final String registrationNumber,
+            @RequestParam final String currentDate)
+            throws IncorrectDateException
+    {
+        Date date = validateDate(currentDate, constants.getTimeFormat());
+
+        if(date != null)
+        {
+            boolean b = parkingSpaceManagement.checkVehicle(registrationNumber, date);
+            return new ResponseEntity<>(b, HttpStatus.OK);
+        }
+        else {
+            throw new IncorrectDateException("Incorrect date format");
+        }
+    }
+
+    @PutMapping(value = URI_STOP_METER + "/{registrationNumber:[0-9]{5}}")
```

Comment:
To `"/{registrationNumber:[0-9]{5}}"` powtarza się dosyć często. Jakbyś chciał zmienić regexpa albo endpoint to będziesz musiał ręcznie szukać wszystkich wystąpień tego stringa

Reply:
Fakt, nie zwróciłem na to uwagi.